### PR TITLE
[12.0] [MIG] product_pricelist_supplierinfo

### PR DESCRIPTION
**Pull request by Nikul-Chaudhary**
Supplier info prices in sales pricelists
========================================
This module allows you to create a sales pricelist based on product supplierinfo prices. If you want, you can bypass minimum quantity in pricelist
item.

### CODE REVIEW

**Comment by Nikul-Chaudhary** [comment]
@OCA/product-maintainers could you please review code?

**Comment by Nikul-Chaudhary** [comment]
@pedrobaeza @cubells @emagdalenaC2i Could you please review it?

**Review by tbaden — APPROVED** [review]
code review: lgtm

**Comment by pedrobaeza** [comment]
Can you improve a bit the test coverage?

**Comment by Nikul-Chaudhary** [comment]
@pedrobaeza @tbaden  Could you please review again?

**Comment by Nikul-Chaudhary** [comment]
@rafaelbn Can you please review it.

**Comment by Nikul-Chaudhary** [comment]
@pedrobaeza @rafaelbn if you don't mind, can you please review it?

File: product_pricelist_supplierinfo/tests/test_product_supplierinfo.py [hunk 1]
```diff
@@ -9,33 +9,31 @@ class TestProductSupplierinfo(common.SavepointCase):
...
-                    'price': 50,
-                }),
-                (0, 0, {
-                    'name': cls.supplier2.id,
-                    'min_qty': 1,
-                    'price': 10,
-                }),
-            ],
-        })
+        cls.product = cls.env.ref('product.product_product_2')
```
Comment (pedrobaeza) [inline product_pricelist_supplierinfo/tests/test_product_supplierinfo.py]: Please don't use demo data. And why are you changing it having already without demo data?

File: product_pricelist_supplierinfo/tests/test_product_supplierinfo.py [hunk 2]
```diff
@@ -9,33 +9,31 @@ class TestProductSupplierinfo(common.SavepointCase):
...
+            }),
+            (0, 0, {
+                'name': cls.supplier2.id,
+                'min_qty': 1,
+                'price': 10,
+            }),
+        ]})
         cls.pricelist = cls.env['product.pricelist'].create({
             'name': 'Supplierinfo Pricelist',
-            'discount_policy': 'without_discount',
```
Comment (pedrobaeza) [inline product_pricelist_supplierinfo/tests/test_product_supplierinfo.py]: Why removing this?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
@Nikul-Chaudhary can you attend my comments?

**Comment by Nikul-Chaudhary** [comment]
@pedrobaeza now ready to merge

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**